### PR TITLE
Add info on query-partition-filter to Hive

### PR DESCRIPTION
**Pull request by rosewms**

### CODE REVIEW

File: docs/src/main/sphinx/connector/hive.rst [hunk 1]
```diff
@@ -384,6 +384,11 @@ Property Name                                      Description
                                                    for tables. It must be either 'skip' or an octal number,
                                                    with a leading 0. If set to 'skip', permissions of newly
                                                    created directories will not be set by Trino.
+
+``hive.query-partition-filter-required``           Requires a filter on a partition key for a query.             ``true``
+                                                   You can set this as a configuration property or as
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive.rst:389]: Use wording like in hive.legacy-hive-view-translation instead.

File: docs/src/main/sphinx/connector/hive.rst [hunk 2]
```diff
@@ -384,6 +384,11 @@ Property Name                                      Description
                                                    for tables. It must be either 'skip' or an octal number,
                                                    with a leading 0. If set to 'skip', permissions of newly
                                                    created directories will not be set by Trino.
+
+``hive.query-partition-filter-required``           Requires a filter on a partition key for a query.             ``true``
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive.rst:388]: This sentence does not capture what the property enables/disables. What does it do?

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/hive.rst (hunk 2 above)
Comment (m57lyra) [reply docs/src/main/sphinx/connector/hive.rst:388]: Do both HAVING and WHERE work as filters?

**Review by m57lyra — APPROVED** [review]
LGTM after clarification

File: docs/src/main/sphinx/connector/hive.rst [hunk 3]
```diff
@@ -384,6 +384,10 @@ Property Name                                      Description
                                                    for tables. It must be either 'skip' or an octal number,
                                                    with a leading 0. If set to 'skip', permissions of newly
                                                    created directories will not be set by Trino.
+
+``hive.query-partition-filter-required``           Force a query to use a partition filter. You can use the     ``true``
```
Comment (phd3) [inline docs/src/main/sphinx/connector/hive.rst:388]: default is false.

**Review by phd3 — COMMENTED** [review]
lgtm % comment. we should also followup with `hive.query-partition-filter-required-schemas`

**Review by jhlodin — APPROVED** [review]

**Comment by phd3** [comment]
merged, thanks!

**Pull request merged**